### DISCUSSION TITLE
Error initializing lotus node "ERROR: initializing node: starting node: could not build arguments for function"

### DISCUSSION THREAD

**Discussion by chenjunjin686637** [filecoin-project/lotus#7393] [General Q&A] [answered]
ERROR: initializing node: starting node: could not build arguments for function "github.com/filecoin-project/lotus/node/modules/lp2p".StartListening.func1 (/data/lotus/node/modules/lp2p/addrs.go:98): failed to build host.Host: could not build arguments for function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:14): failed to build lp2p.BaseIpfsRouting: could not build arguments for function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:14): failed to build dtypes.NetworkName: could not build arguments for function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:14): failed to build dtypes.AfterGenesisSet: received non-nil error from function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:14): genesis in the repo is not the one expected by this version of Lotus!  What causes this?

**Comment by de-authority** [comment, marked as answer]
You have several versions mixed together？ see this thread: https://filecoinproject.slack.com/archives/CEGN061C5/p1597411916060100
